ksql-common/src/test/java/io/confluent/ksql/util/timestamp/TimestampExtractionPolicyFactoryTest.java

```diff
@@ -32,14 +40,101 @@
   private final SchemaBuilder schemaBuilder = SchemaBuilder.struct()
       .field("id", Schema.OPTIONAL_INT64_SCHEMA);
 
+  private KsqlConfig ksqlConfig;
+
+  @Before
+  public void setup() {
+    ksqlConfig = new KsqlConfig(Collections.emptyMap());
+  }
+
   @Test
   public void shouldCreateMetadataPolicyWhenTimestampFieldNotProvided() {
     // When:
     final TimestampExtractionPolicy result = TimestampExtractionPolicyFactory
-        .create(KsqlSchema.of(schemaBuilder.build()), Optional.empty(), Optional.empty());
+        .create(
+            ksqlConfig,
+            KsqlSchema.of(schemaBuilder.build()),
+            Optional.empty(),
+            Optional.empty()
+        );
+
+    // Then:
+    assertThat(result, instanceOf(MetadataTimestampExtractionPolicy.class));
+    assertThat(result.create(0), instanceOf(FailOnInvalidTimestamp.class));
+  }
+
+  @Test(expected = ConfigException.class)
```

Comment:
Why not use `ExpectedException` so you can also check the correctness of the error messages?

Reply:
I followed how other tests cases of this class verify error exceptions. 
Btw, What is our preference? Should I modify the other tests cases with the `ExpectdException`?

Reply:
`ExpectdException` would be preferred. Check `KsqlRequestTest` for examples.

Reply:
Done